src/rosdistro/rosdistro.py

```diff
@@ -198,25 +198,26 @@ def __init__(self, name, repository):
     def _fetch_package_xml(self, rosdistro):
         repo = self.repository
         if 'github.com' in repo.url:
-            url = repo.url
-            upstream_version = repo.version.split('-')[0]
-            release_tag = 'release/{0}/{1}'.format(self.name, upstream_version)
-            url = url.replace('.git', '/{0}/package.xml'.format(release_tag))
-            url = url.replace('git://', 'https://')
-            url = url.replace('https://', 'https://raw.')
             try:
                 try:
-                    package_xml = urlopen(url).read()
-                except Exception as e:
-                    msg = "Failed to read package.xml file from url '{0}': {1}".format(url, e)
-                    warning(msg)
                     url = repo.url
                     release_tag = 'release/{0}/{1}/{2}'.format(rosdistro, self.name, repo.version)
                     tail = '/{0}/package.xml'.format(release_tag)
                     url = url.replace('.git', tail)
                     url = url.replace('git://', 'https://')
                     url = url.replace('https://', 'https://raw.')
-                    info("Trying to read from url '{0}' instead".format(url))
+                    package_xml = urlopen(url).read()
+                except Exception as e:                    
+                    msg = "Failed to read package.xml file from url '{0}': {1}".format(url, e)
+                    warning(msg)
+                    url = repo.url
+                    upstream_version = repo.version.split('-')[0]
+                    release_tag = 'release/{0}/{1}'.format(self.name, upstream_version)
+                    url = url.replace('.git', '/{0}/package.xml'.format(release_tag))
+                    url = url.replace('git://', 'https://')
+                    url = url.replace('https://', 'https://raw.')
```

Comment:
This seems like a duplication of the more robust logic which is here:

https://github.com/ros-infrastructure/rosdistro/blob/master/src/rosdistro/manifest_provider/github.py#L65

Can we consolidate?